src/Extension/IntlExtension.php

```diff
@@ -0,0 +1,393 @@
+    {
+        if (!class_exists(Timezones::class)) {
+            throw new \LogicException('The "timezone_name" filter requires the symfony/intl package to be installed.');
+        }
+
+        return Timezones::getName($timezone, $locale);
+    }
+
+    public function getCountryTimezones(string $country): array
+    {
+        if (!class_exists(Timezones::class)) {
+            throw new \LogicException('The "country_timezone" filter requires the symfony/intl package to be installed.');
+        }
+
+        return Timezones::forCountryCode($country);
+    }
+
+    public function formatCurrency($amount, string $currency, string $locale = null, array $attrs = []): string
+    {
+        if (!class_exists('Locale')) {
+            throw new RuntimeException('The "format_currency" filter requires the PHP intl extension or the symfony/intl package to be installed.');
+        }
+
+        $formatter = $this->createNumberFormatter($locale, 'currency', $attrs);
+
+        if (false === $ret = $formatter->formatCurrency($amount, $currency)) {
+            throw new RuntimeError('Unable to format the given number as a currency.');
+        }
+
+        return $ret;
+    }
+
+    public function formatNumber($number, string $locale = null, array $attrs = [], string $style = 'decimal', string $type = 'default'): string
+    {
+        if (!class_exists('Locale')) {
+            throw new RuntimeException('The "format_number" filter requires the PHP intl extension or the symfony/intl package to be installed.');
+        }
+
+        if (!isset(self::NUMBER_TYPES[$type])) {
+            throw new RuntimeError(sprintf('The type "%s" does not exist, known types are: "%s".', $type, implode('", "', array_keys(self::NUMBER_TYPES))));
+        }
+
+        $formatter = $this->createNumberFormatter($locale, $style, $attrs);
+
+        if (false === $ret = $formatter->format($number, self::NUMBER_TYPES[$type])) {
+            throw new RuntimeError('Unable to format the given number.');
+        }
+
+        return $ret;
+    }
+
+    public function formatNumberStyle(string $style, $number, string $locale = null, array $attrs = [], string $type = 'default'): string
+    {
+        return $this->formatNumber($number, $locale, $attrs, $style, $type);
+    }
+
+    /**
+     * @param \DateTimeInterface|string|null  $date     A date or null to use the current time
+     * @param \DateTimeZone|string|false|null $timezone The target timezone, null to use the default, false to leave unchanged
+     */
+    public function formatDateTime(Environment $env, $date, string $locale = null, ?string $dateFormat = 'medium', ?string $timeFormat = 'medium', string $pattern = '', $timezone = null, string $calendar = 'gregorian'): string
+    {
+        if (!class_exists('Locale')) {
+            throw new RuntimeException('The "format_date" filter requires the PHP intl extension or the symfony/intl package to be installed.');
```

Comment:
I think you want to recommend this package instead https://github.com/symfony/polyfill/tree/master/src/Intl/Icu
That the polyfill is in symfony/intl is meant to be an implementation detail that might change.

Reply:
Right,  changed